extensions-contrib/tdigestsketch/src/main/java/org/apache/druid/query/aggregation/tdigestsketch/sql/TDigestGenerateSketchSqlAggregator.java

```diff
@@ -0,0 +1,194 @@
+        if (factory instanceof TDigestSketchAggregatorFactory) {
+          final TDigestSketchAggregatorFactory theFactory = (TDigestSketchAggregatorFactory) factory;
+
+          // Check input for equivalence.
+          final boolean inputMatches;
+          final VirtualColumn virtualInput = existing.getVirtualColumns()
+                                                     .stream()
+                                                     .filter(
+                                                         virtualColumn ->
+                                                             virtualColumn.getOutputName()
+                                                                          .equals(theFactory.getFieldName())
+                                                     )
+                                                     .findFirst()
+                                                     .orElse(null);
+
+          if (virtualInput == null) {
+            inputMatches = input.isDirectColumnAccess()
+                           && input.getDirectColumn().equals(theFactory.getFieldName());
+          } else {
+            inputMatches = ((ExpressionVirtualColumn) virtualInput).getExpression()
+                                                                   .equals(input.getExpression());
+          }
+          final boolean matches = inputMatches && theFactory.getCompression() == compression;
+
+          if (matches) {
+            // Found existing one. Use this.
+            return Aggregation.create(
+                theFactory
+            );
+          }
+        }
+      }
+    }
+
+    // No existing match found. Create a new one.
+    final List<VirtualColumn> virtualColumns = new ArrayList<>();
+
+    if (input.isDirectColumnAccess()) {
+      aggregatorFactory = new TDigestSketchAggregatorFactory(
+          aggName,
+          input.getDirectColumn(),
+          compression
+      );
+    } else {
+      VirtualColumn virtualColumn = virtualColumnRegistry.getOrCreateVirtualColumnForExpression(
+          plannerContext,
+          input,
+          SqlTypeName.FLOAT
+      );
+      virtualColumns.add(virtualColumn);
+      aggregatorFactory = new TDigestSketchAggregatorFactory(
+          aggName,
+          virtualColumn.getOutputName(),
+          compression
+      );
+    }
+
+    return Aggregation.create(
+        virtualColumns,
+        aggregatorFactory
+    );
+  }
+
+  private static class TDigestGenerateSketchSqlAggFunction extends SqlAggFunction
+  {
+    //private static final String SIGNATURE1 = "'" + NAME + "(column)'\n";
+    private static final String SIGNATURE2 = "'" + NAME + "(column, compression)'\n";
+
+    TDigestGenerateSketchSqlAggFunction()
+    {
+      super(
+          NAME,
+          null,
+          SqlKind.OTHER_FUNCTION,
+          ReturnTypes.explicit(SqlTypeName.OTHER),
+          null,
+          OperandTypes.and(
```

Comment:
The agg function here should be adjusted to support the optional compression param like in the quantile version